frontend/src/authenticated/EditPost.js

```diff
@@ -28,16 +34,32 @@ const EditPost = props => {
 
       setMarkdown(post.body);
       setTitle(post.title);
+      setCategoriesIds(post.categoriesIds);
     };
 
     fetchPost();
   }, [id]);
 
+  useEffect(() => {
+    setCategoriesOptions(convertToSelectOptions(allCategories));
+  }, [allCategories]);
+
+  useEffect(() => {
+    const userCategories = allCategories.filter(category =>
+      categoriesIds.includes(category.id)
+    );
+    const userCategoriesOptions = convertToSelectOptions(
+      userCategories
+    );
+
+    setUserCategoriesOptions(userCategoriesOptions);
+  }, [allCategories, categoriesIds]);
```

Comment:
keep it in one useEffect :-)

Reply:
There would be too many variables in dependency array, causing infinite loop. I tried many approaches and this one works.